hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -317,6 +317,9 @@ int SaraNcpClient::initParser(Stream* stream) {
         // self->checkRegistrationState();
         // Cellular Global Identity (partial)
         if (r >= 3) {
+            if(r >= 4 && val[3] == 0xFFFFFFFF) {
```

Comment:
since `val` is unsigned, the sscanf of -1 will result in `0xFFFFFFFF` Should we change `val` to signed int to make the comparison to -1 more obvious? Or leave as is?

Reply:
Hmm, good question.  I think the other values that val[] is capturing would potentially be messier if val[] was changed to an int.  So possibly add a comment that we are checking for `RAT == -1` on that line.

Reply:
Made comment hopefully clearer